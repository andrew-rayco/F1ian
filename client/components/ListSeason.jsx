import React from 'react'

import { Link } from 'react-router-dom'

const ListSeason = (props) => {
  let year = props.year
  return (
    <li className="season">
      <Link to={`/season/${year}`}>{year}</Link>
    </li>
  )
}

export default ListSeason
