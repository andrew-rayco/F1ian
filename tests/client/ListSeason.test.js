import React from 'react'
import { shallow } from 'enzyme'

import ListSeason from '../../client/components/ListSeason'

describe('ListSeason component', () => {
  const app = shallow(<ListSeason />)

  it('renders correctly', () => {
    expect(app).toMatchSnapshot()
  })

  it('contains a single .season div', () => {
    expect(app.find('.season').length).toEqual(1)
  })

  it('contains a link', () => {
    expect(app.find('Link').exists()).toBe(true)
  })
})
