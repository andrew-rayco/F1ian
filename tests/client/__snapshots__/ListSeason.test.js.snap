// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListSeason component renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ListSeason />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Link
        replace={false}
        to="/season/undefined"
      />,
      "className": "season",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": undefined,
        "replace": false,
        "to": "/season/undefined",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Link
          replace={false}
          to="/season/undefined"
        />,
        "className": "season",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": undefined,
          "replace": false,
          "to": "/season/undefined",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
